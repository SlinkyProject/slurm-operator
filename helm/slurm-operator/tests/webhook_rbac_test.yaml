# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
---
suite: test webhook rbac
templates:
- webhook/rbac.yaml
chart:
  version: 1.2.3
  appVersion: 1.2.3
release:
  name: test-release
  namespace: test-namespace

tests:
- it: manifest should match snapshot
  asserts:
  - matchSnapshot: {}

- it: should not render when disabled
  set:
    webhook:
      enabled: false
  asserts:
  - hasDocuments:
      count: 0

- it: should not render when serviceAccount.create is false
  set:
    webhook:
      serviceAccount:
        create: false
  asserts:
  - hasDocuments:
      count: 0


- it: should set namespace
  documentSelector:
    path: kind
    value: ServiceAccount
  set:
    namespaceOverride: custom-namespace
  asserts:
  - equal:
      path: metadata.namespace
      value: custom-namespace


- it: should set serviceAccount name
  documentSelector:
    path: kind
    value: ServiceAccount
  set:
    webhook:
      serviceAccount:
        name: custom-service-account
  asserts:
  - equal:
      path: metadata.name
      value: custom-service-account
