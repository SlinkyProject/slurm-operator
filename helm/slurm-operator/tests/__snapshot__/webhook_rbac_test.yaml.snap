manifest should match snapshot:
  1: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator-webhook
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: slurm-operator-webhook
      namespace: test-namespace
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator-webhook
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: slurm-operator-webhook
      namespace: test-namespace
    rules:
      - apiGroups:
          - slinky.slurm.net
        resources:
          - accountings
          - controllers
          - loginsets
          - nodesets
          - restapis
          - tokens
        verbs:
          - create
          - delete
          - update
      - apiGroups:
          - ""
        resources:
          - configmaps
        verbs:
          - get
          - list
          - watch
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator-webhook
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: slurm-operator-webhook
      namespace: test-namespace
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: slurm-operator-webhook
    subjects:
      - kind: ServiceAccount
        name: slurm-operator-webhook
        namespace: test-namespace
